x-pack/plugins/infra/server/plugin.ts

```diff
@@ -298,6 +308,8 @@ export class InfraServerPlugin
       savedObjects: core.savedObjects,
     });
 
+    initInfraServer(this.libs, core, pluginsStart);
```

Comment:
I think the route registration should happen in setup lifecycle. https://www.elastic.co/guide/en/kibana/current/kibana-platform-plugin-api.html#plugin-lifecycles 

What do we get from registering the route in the start, also are there any side effects of this change?

Reply:
I understand that's an indented architecture, but it kind of makes things more convoluted in practice where routes still need to access start dependencies. What do you think about the current approach with the `getStartServices()` method? Maybe it bothers only me 😅

Reply:
> What do we get from registering the route in the start, also are there any side effects of this change?

Main benefit is a more straightforward rote dependecies, without a need for `getStartServices()` that's being passed from the `setup` phase. I've tried to address it in [this reply](https://github.com/elastic/kibana/pull/176612#discussion_r1489327489) as well. And I have noticed any side-effect, from what I understand `setup`/`start`/`stop` phases are just a convention and it still boils down to a normal Node server, so it should not matter where we initialize the routes.